api/src/com/cloud/agent/api/storage/OVFHelper.java

```diff
@@ -0,0 +1,333 @@
+            String keepdisk = null;
+            for (int i = 0; i < disks.getLength(); i++) {
+                Element disk = (Element)disks.item(i);
+                String fileRef = disk.getAttribute("ovf:fileRef");
+                if (keepfile == null) {
+                    s_logger.info("FATAL: OVA format error");
+                } else if (keepfile.equals(fileRef)) {
+                    s_logger.info("MDOVA rewriteOVFFile keeping disk = " + fileRef);
+                    keepdisk = disk.getAttribute("ovf:diskId");
+                } else {
+                    s_logger.info("MDOVA rewriteOVFFile removing disk = " + fileRef);
+                    s_logger.info("MDOVA rewriteOVFFile id = " + disk.getAttribute("ovf:diskId"));
+                    toremove.add(disk);
+                }
+            }
+            for (int k = 0; k < items.getLength(); k++) {
+                Element item = (Element)items.item(k);
+                NodeList cn = item.getChildNodes();
+                for (int l = 0; l < cn.getLength(); l++) {
+                    if (cn.item(l) instanceof Element) {
+                        Element el = (Element)cn.item(l);
+                        if ("rasd:HostResource".equals(el.getNodeName())
+                                && !(el.getTextContent().contains("ovf:/file/" + keepdisk) || el.getTextContent().contains("ovf:/disk/" + keepdisk))) {
+                            s_logger.info("MDOVA to remove " + el.getTextContent());
+                            toremove.add(item);
+                            break;
+                        }
+                    }
+                }
+            }
+
+            for (Element rme : toremove) {
+                s_logger.info("MDOVA remove " + rme.getTagName());
+                if (rme.getParentNode() != null) {
+                    rme.getParentNode().removeChild(rme);
+                }
+            }
+
+            final StringWriter writer = new StringWriter();
+            final StreamResult result = new StreamResult(writer);
+            final TransformerFactory tf = TransformerFactory.newInstance();
+            final Transformer transformer = tf.newTransformer();
+            final DOMSource domSource = new DOMSource(doc);
+            transformer.transform(domSource, result);
+            PrintWriter outfile = new PrintWriter(newOvfFilePath);
+            outfile.write(writer.toString());
+            outfile.close();
+        } catch (SAXException | IOException | ParserConfigurationException | TransformerException e) {
+            s_logger.info("Unexpected exception caught while removing network elements from OVF:" + e.getMessage(), e);
+            throw new CloudRuntimeException(e);
+        }
+    }
+
+    OVFDisk getDisk(String fileRef, List<OVFDisk> disks) {
+        for (OVFDisk disk : disks) {
+            if (disk._fileRef.equals(fileRef)) {
+                return disk;
+            }
+        }
+        return null;
+    }
+
+    class OVFFile {
```

Comment:
All of these can be generated from a Schema using something like JAXB or similar tools, and then the parsing tool can generate the objects ready to use for you.